packages/adapter-drizzle/src/pg/index.ts

```diff
@@ -0,0 +1,238 @@
+import {
+  integer,
+  timestamp,
+  pgTable,
+  text,
+  primaryKey,
+} from "drizzle-orm/pg-core"
+import { drizzle } from "drizzle-orm/postgres-js"
+import postgres from "postgres"
+import { Adapter, AdapterAccount } from '@auth/core/adapters'
+import { and, eq } from "drizzle-orm"
+
+const queryConnection = postgres(process.env.DATABASE_URL!)
+
+export const users = pgTable("users", {
+  id: text("id").notNull().primaryKey(),
+  name: text("name"),
+  email: text("email").notNull(),
+  emailVerified: timestamp("emailVerified", { mode: "date" }),
+  image: text("image"),
+})
+
+export const accounts = pgTable(
+  "accounts",
+  {
+    userId: text("userId")
+      .notNull()
+      .references(() => users.id, { onDelete: "cascade" }),
+    type: text("type").$type<AdapterAccount["type"]>().notNull(),
+    provider: text("provider").notNull(),
+    providerAccountId: text("providerAccountId").notNull(),
+    refresh_token: text("refresh_token"),
+    access_token: text("access_token"),
+    expires_at: integer("expires_at"),
+    token_type: text("token_type"),
+    scope: text("scope"),
+    id_token: text("id_token"),
+    session_state: text("session_state"),
+  },
+  (account) => ({
+    compoundKey: primaryKey(account.provider, account.providerAccountId),
+  })
+)
+
+export const sessions = pgTable("sessions", {
+  sessionToken: text("sessionToken").notNull().primaryKey(),
+  userId: text("userId")
+    .notNull()
+    .references(() => users.id, { onDelete: "cascade" }),
+  expires: timestamp("expires", { mode: "date" }).notNull(),
+})
+
+export const verificationTokens = pgTable(
+  "verificationToken",
+  {
+    identifier: text("identifier").notNull(),
+    token: text("token").notNull(),
+    expires: timestamp("expires", { mode: "date" }).notNull(),
+  },
+  (vt) => ({
+    compoundKey: primaryKey(vt.identifier, vt.token),
+  })
+)
+
+export const db = drizzle(queryConnection)
+export type DbClient = typeof db
+
+export const defaultSchema = { users, accounts, sessions, verificationTokens }
+export type DefaultSchema = typeof defaultSchema
+interface CustomSchema extends DefaultSchema { }
+
+export function pgDrizzleAdapter(
+  client: DbClient,
+  schema?: Partial<CustomSchema>
+): Adapter {
+  const { users, accounts, sessions, verificationTokens } = {
+    users: schema?.users ?? defaultSchema.users,
+    accounts: schema?.accounts ?? defaultSchema.accounts,
+    sessions: schema?.sessions ?? defaultSchema.sessions,
+    verificationTokens:
+      schema?.verificationTokens ?? defaultSchema.verificationTokens,
+  }
+
+  return {
+    createUser: async (data) => {
+      return client
+        .insert(users)
+        .values({ ...data, id: crypto.randomUUID() })
+        .returning()
+        .then((res) => res[0])
```

Comment:
This is giving me a typescript error when trying to pass the adapter to `nextauth` as Adapter

```
Type 'Adapter' is not assignable to type 'Adapter | undefined'.
  Type 'Adapter' is not assignable to type 'DefaultAdapter & { createVerificationToken: (verificationToken: VerificationToken) => Awaitable<VerificationToken | null | undefined>; useVerificationToken: (params: { ...; }) => Awaitable<...>; }'.
    Type 'Adapter' is not assignable to type 'DefaultAdapter'.
      Types of property 'createUser' are incompatible.
        Type '((user: Omit<AdapterUser, "id">) => Awaitable<AdapterUser>) | undefined' is not assignable to type '(user: Omit<AdapterUser, "id">) => Awaitable<AdapterUser>'.
          Type 'undefined' is not assignable to type '(user: Omit<AdapterUser, "id">) => Awaitable<AdapterUser>'.

```